test/src/util.ts

```diff
@@ -18,20 +18,45 @@ import * as path from 'path'
 import * as vscode from 'vscode'
 
 /**
- * Activates the extension in the `testWorkspace` directory.
+ * Activates the extension and copies the contents of the given test workspace directory into the extive workspace.
+ *
+ * @param testWorkspaceName The name of the workspace directory to copy, e.g. `codeActions`.
  */
-export async function activate() {
+export async function activate(testWorkspaceName: string) {
   // The extensionId is `publisher.name` from package.json
   const ext = vscode.extensions.getExtension('flix.flix')
   if (ext === undefined) {
     throw new Error('Failed to activate extension')
   }
 
+  await copyWorkspace(testWorkspaceName)
+
   // This includes the time it takes for the compiler to download
   // The time it takes for the compiler to start will be awaited in the first command sent to the extension
   await ext.activate()
 }
 
+/**
+ * Clears the test workspace, and copies the contents of the given test workspace directory into the active workspace.
+ *
+ * @param testWorkspaceName The name of the workspace directory to copy, e.g. `codeActions`.
+ */
+async function copyWorkspace(testWorkspaceName: string) {
+  vscode.commands.executeCommand('workbench.action.closeAllEditors')
+
+  const activeWorkspaceUri = vscode.workspace.workspaceFolders![0].uri
+  const contents = await vscode.workspace.fs.readDirectory(activeWorkspaceUri)
+  const names = contents.map(([name, _]) => name)
+
+  // Delete all files except .gitkeep and flix.jar
+  const namesToDelete = names.filter(name => name !== '.gitkeep' && name !== 'flix.jar')
+  const urisToDelete = namesToDelete.map(name => vscode.Uri.joinPath(activeWorkspaceUri, name))
```

Comment:
I am always very careful with file deletion. Would it be sensible to require the file name to end in .flix? Or at list of extensions?

Reply:
Well, if the tests should be reliable the active workspace needs to be reset entirely, including flix.toml and all packages which may have been downloaded. The only exception I would think of as making sense is flix.jar, like it is now.

I can't immediately think of any other cases where allowing other files to survive would be anything other than confusing and error prone. Do you have anything specific in mind?

Reply:
What I mean is: lets try to be careful about what we delete. 

When a path goes haywire its better to only be deleting e.g. Flix files. 

Even better could we track the files we are supposed to delete?